packages/p2p/src/components/buy-sell/buy-sell-modal.jsx

```diff
@@ -67,19 +71,35 @@ const BuySellModal = ({ table_type, selected_ad, should_show_popup, setShouldSho
             setErrorMessage(null);
         }
 
-        my_profile_store.setShouldShowAddPaymentMethodForm(false);
         my_profile_store.setSelectedPaymentMethod('');
         my_profile_store.setSelectedPaymentMethodDisplayName('');
 
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [should_show_popup]);
 
+    const generateModalTitle = () => {
+        if (my_profile_store.should_show_add_payment_method_form) {
+            if (isDesktop()) {
+                return (
+                    <>
+                        <Icon
+                            icon='IcArrowLeftBold'
+                            onClick={() => my_profile_store.setIsCancelAddPaymentMethodModalOpen(true)}
+                        />
+                        &nbsp;{localize('Add payment method')}
+                    </>
+                );
+            }
+            return localize('Add payment method');
+        }
+        if (table_type === buy_sell.BUY) {
+            return localize('Buy {{ currency }}', { currency: selected_ad.account_currency });
+        }
+        return localize('Sell {{ currency }}', { currency: selected_ad.account_currency });
+    };
+
     const Form = general_store.nickname ? BuySellForm : NicknameForm;
-    const modal_title = my_profile_store.should_show_add_payment_method_form
-        ? localize('Add payment method')
-        : table_type === buy_sell.BUY
-        ? localize('Buy {{ currency }}', { currency: selected_ad.account_currency })
-        : localize('Sell {{ currency }}', { currency: selected_ad.account_currency });
+    const modal_title = generateModalTitle();
```

Comment:
We don't need an extra variable for this, you can call it where `modal_title` is used.